box.ctl: add box.ctl.wal_sync() to wait flush (#10422)
tarantool/tarantool: changelogs/unreleased/gh-tarantool-10142-box-ctl-wal-sync-lua.md

> ## feature/lua

Author: Fixed apparently but your opinion on that is needed

> * Added `box.ctl.wal_sync()` function which uses C `wal_sync()` and wait until
>   submitted writes are flushed to the disk (gh-10142).

Reviewer: ```suggestion
* Added `box.ctl.wal_sync()` function which uses C `wal_sync()` and waits until
  submitted writes are flushed to the disk (gh-10142).
```

Author: Thanks! I fxed that but then I actually rewrote most of that changelog

Reviewer 2: @lenkis, please, chech the changelog one more time. It was significantly changed

## feature/lua

* Added `box.ctl.wal_sync()` function which waits until
  all submitted writes are successfully flushed to the disk.

>   Throws an error if write failed. The main goal is to

Author: Fixed

> * Added `box.ctl.wal_sync()` function which waits until
>   all submitted writes are successfully flushed to the disk.
>   Throws an error if write fails. The main goal is to
>   provide a way to synchronize vclock between instances
>   primarily for switching masters (gh-10142).

Reviewer 3: ```suggestion
* Added the `box.ctl.wal_sync()` function, which waits until
  all submitted writes are successfully flushed to the disk.
  Throws an error if a write fails. The function provides a way
  to synchronize vclock between instances when switching masters
  (gh-10142).
```

>   primarily for switching masters (gh-10142).

Reviewer 4: IMO the changelog and doc request are misleading.

In doc request:
> It is primarily needed for synchronisation between
old and new masters or actually any other situations when user
need consistent vclock between instances.

In changelog:
> The main goal is to
  provide a way to synchronize vclock between instances
  primarily for switching masters 

Having read these, I'd believe that `wal_sync` somehow does a synchronization of vclocks between instances.

Let's better rephrase these places in both doc request and changelog. For example, `After the function is executed one may reliably use box.info.vclock for comparisons when choosing a new master`.

Reviewer 4: Also the changelog should go to `feature/core`, as it doesn't have anything to do with lua itself.
(`core` and not `box` since we've agreed to merge these two paragraphs. For example, see the latest release changelogs https://github.com/tarantool/tarantool/releases)

Author: Sure. I have no objections to the proposed fix so I added it. I hope it's more clear now what the function does. Fixed.
